crates/walrus-service/src/client/cli/args.rs

```diff
@@ -266,6 +266,122 @@ pub enum CliCommands {
+    ReadQuilt {
+        /// The quilt ID, which is the BlobID of the quilt.
+        ///
+        /// It is required unless `--quilt-patch-id` is used.
+        #[serde_as(as = "Option<DisplayFromStr>")]
+        #[arg(
+            long,
+            allow_hyphen_values = true,
+            value_parser = parse_blob_id,
+            required_unless_present = "quilt_patch_ids",
+            help_heading = "Arguments",
+        )]
+        quilt_id: Option<BlobId>,
+
+        /// The identifiers to read from the quilt.
+        ///
+        /// It is required to be used with `--quilt-id`.
+        ///
+        /// Example:
+        /// walrus read-quilt --quilt-id `<ID>` --identifier `<IDENTIFIER>`...
+        #[arg(
+            long = "identifier",
+            conflicts_with = "tag",
+            conflicts_with = "quilt_patch_ids",
+            help_heading = "Arguments"
+        )]
+        #[serde(default)]
+        identifiers: Vec<String>,
+
+        /// The tag key.
+        ///
+        /// It is required to be used with `--quilt-id` and `--value`.
+        ///
+        /// Example:
+        /// walrus read-quilt --quilt-id `<ID>` --tag `<KEY>` --value `<VALUE>`
+        #[arg(
+            long,
+            requires = "value",
+            conflicts_with = "identifiers",
+            conflicts_with = "quilt_patch_ids",
+            help_heading = "Arguments"
+        )]
+        #[serde(default)]
+        tag: Option<String>,
+
+        /// The tag value to match.
+        ///
+        /// It is required to be used with `--quilt-id` and `--tag`.
+        ///
+        /// Example:
+        /// walrus read-quilt --quilt-id `<ID>` --tag `<KEY>` --value `<VALUE>`
+        #[arg(
+            long,
+            requires = "tag",
+            conflicts_with = "identifiers",
+            conflicts_with = "quilt_patch_ids",
+            help_heading = "Arguments"
+        )]
+        #[serde(default)]
+        value: Option<String>,
+
+        /// The quilt patch IDs.
+        ///
+        /// It should be used alone without other arguments.
+        ///
+        /// Example:
+        /// walrus read-quilt --quilt-patch-id `<PATCH_ID>`...
+        #[serde_as(as = "Vec<DisplayFromStr>")]
+        #[arg(
+            long = "quilt-patch-id",
+            alias = "patch-id",
+            allow_hyphen_values = true,
+            value_parser = parse_quilt_patch_id,
+            action = clap::ArgAction::Append,
+            conflicts_with = "quilt_id",
+            conflicts_with = "identifiers",
+            conflicts_with = "tag",
+            help_heading = "Arguments",
+        )]
+        #[serde(default)]
+        quilt_patch_ids: Vec<QuiltPatchId>,
+
+        /// The directory path where to write the quilt patches.
+        /// The blobs are written to the directory with the same name as the identifier.
+        /// The metadata of the quilt patches, including identifiers and tags are printed to the
```

Comment:
Question: does the metadata mean `tag`? If not the same as the blob metadata storing the merkle tree, maybe use a different word to avoid confusion.